packages/adapters/watcher/src/adapter.ts

```diff
@@ -8,30 +9,48 @@ import {
   AssetInfo,
   Report,
   PauseResponse,
-  VerifyResponse,
   WatcherAlertsConfig,
+  WatcherInvariantResponse,
 } from "./types";
 import { AssetVerifier } from "./verifiers";
 
 // Aggregation class for interfacing with all adapter functionality.
 export class WatcherAdapter {
   private readonly verifiers: Verifier[];
   private readonly pauser: Pauser;
+  private readonly context: VerifierContext;
+
+  // transactions sent to connext within the polling interval. cleared after
+  // each time `checkInvariants` is called
+  private readonly latestTransactions: Record<string, string[]> = {};
+
+  private push = false;
+
+  // this is set when `checkInvariants` determines that the system needs to be
+  // paused. will hold all transactions between checks that have gone to
+  // connext
+  private transactionSnapshot: Record<string, string[]> | undefined;
 
   constructor(context: VerifierContext, assets: AssetInfo[]) {
+    this.context = context;
     // Should set all applicable verifiers here!
     this.verifiers = [new AssetVerifier(context, assets)];
     this.pauser = new Pauser(context);
+    this.beginRecordingTransactions();
```

Comment:
Feels slightly dangerous, as we'll basically be piling on txs indefinitely until the consumer calls `checkInvariants`...

Right now, we know the consumer will call `checkInvariants` frequently enough that this should cause no concern... anyway, just making a mental note for myself here about this. We could make sure we're fixing this problem by capping the number of txs really high so we have a built-in failsafe in case we start consuming a lot of memory

Reply:
this establishes the listeners, but the transactions are cleared between non-alerting intervals. 

can change to between all intervals as there are the snapshots

Reply:
but keep in mind the memory buildup would only happen after there had been an alert, so wouldn't really be top of mind

Reply:
cleared after each interval